src/ledger/LedgerTxn.cpp

```diff
@@ -168,11 +168,90 @@ LedgerEntryPtr::isDeleted() const
 {
     return mState == EntryPtrState::DELETED;
 }
+bool
+LedgerKeyMeter::canLoad(LedgerKey const& key, size_t entrySizeBytes) const
+{
+    return maxReadQuotaForKey(key) >= entrySizeBytes;
+}
+
+void
+LedgerKeyMeter::addTxn(uint32_t readQuota, UnorderedSet<LedgerKey>& keys)
+{
+    txReadBytes.push_back(readQuota);
+    auto txId = txReadBytes.size() - 1;
+    for (auto const& key : keys)
+    {
+        if (key.type() == TTL)
+        {
+            // TTL entries do not count towards the txn read byte limit.
+            continue;
+        }
+        ledgerKeyToTxs[key].insert(txId);
+    }
+}
+
+void
+LedgerKeyMeter::updateReadQuotasForKey(LedgerKey const& key,
+                                       size_t entrySizeBytes)
+{
+    if (key.type() == TTL)
+    {
+        // TTL entries do not count towards the txn read byte limit.
+        return;
+    }
+    auto iter = ledgerKeyToTxs.find(key);
+    // This assert will fail if the transaction is not metered (classic).
+    // Classic prefetching should never call this function.
+    releaseAssert(iter != ledgerKeyToTxs.end());
+    // Update the read quota for every transaction containing this key.
+    bool exceedsQuotaForAllTxns = true;
+    for (auto txn : iter->second)
+    {
+        if (txReadBytes[txn] < entrySizeBytes)
```

Comment:
`txReadBytes.at(txn)` instead of `[]`. Could do something like `auto& readQuota = txReadBytes.at(txn)` and use readQuota everywhere else.